components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -51,23 +54,28 @@ type NatsBackend interface {
 }
 
 type Nats struct {
-	config            env.NatsConfig
-	defaultSubsConfig env.DefaultSubscriptionConfig
-	logger            *logger.Logger
-	client            cev2.Client
-	connection        *nats.Conn
-	subscriptions     map[string]*nats.Subscription
-	sinks             sync.Map
-	connClosedHandler ConnClosedHandler
+	config                     env.NatsConfig
+	defaultSubsConfig          env.DefaultSubscriptionConfig
+	logger                     *logger.Logger
+	client                     cev2.Client
+	connection                 *nats.Conn
+	subscriptions              map[string]*nats.Subscription
+	sinks                      sync.Map
+	connClosedHandler          ConnClosedHandler
+	subscriptionDeliveryMetric *prometheus.CounterVec
 }
 
 func NewNats(config env.NatsConfig, subsConfig env.DefaultSubscriptionConfig, logger *logger.Logger) *Nats {
-	return &Nats{
-		config:            config,
-		defaultSubsConfig: subsConfig,
-		logger:            logger,
-		subscriptions:     make(map[string]*nats.Subscription),
+	n := &Nats{
+		config:                     config,
+		defaultSubsConfig:          subsConfig,
+		logger:                     logger,
+		subscriptions:              make(map[string]*nats.Subscription),
+		subscriptionDeliveryMetric: NewDeliverPerSubscriptionMetric(),
 	}
+	metrics.Registry.Unregister(n.subscriptionDeliveryMetric)
```

Comment:
I need to unregister the metric, because it might happen, that the metric will be registered more then once, this will throw an error. Possible scenarios for multiple registering:
1) switching the backend to beb and then to nats.
1) calling unit/integration tests, which call the `NewNats` function multiple times.